File: CarHomeWidget/src/main/java/com/anod/car/home/prefs/model/InCarInterface.java
package com.anod.car.home.prefs.model;

import android.content.ComponentName;
import androidx.collection.ArrayMap;

/**
 * @author algavris
 * @date 08/04/2016.
 */
public interface InCarInterface {
    String BRIGHTNESS_DISABLED = "disabled";
    String BRIGHTNESS_AUTO = "auto";
    String BRIGHTNESS_DAY = "day";
    String BRIGHTNESS_NIGHT = "night";
    String AUTOANSWER_DISABLED = "disabled";
    String AUTOANSWER_IMMEDIATLY = "immediately";
    String AUTOANSWER_DELAY_5 = "delay-5";
    String WIFI_NOACTION = "no_action";
    String WIFI_TURNOFF = "turn_off_wifi";
    String WIFI_DISABLE = "disable_wifi";
    int DEFAULT_VOLUME_LEVEL = 80;

    boolean isInCarEnabled();

    void setInCarEnabled(boolean enabled);

    boolean isAutoSpeaker();

    ArrayMap<String, String> getBtDevices();

    void setBtDevices(ArrayMap<String, String> btDevices);

    boolean isPowerRequired();

    boolean isHeadsetRequired();

    boolean isBluetoothRequired();

    boolean isDisableBluetoothOnPower();

    boolean isEnableBluetoothOnPower();

    boolean isDisableScreenTimeout();

    boolean isAdjustVolumeLevel();

    boolean isActivityRequired();

    int getMediaVolumeLevel();

    int getCallVolumeLevel();

    boolean isEnableBluetooth();

    String getBrightness();

    void setPowerRequired(boolean powerRequired);

    void setHeadsetRequired(boolean headsetRequired);

    void setActivityRequired(boolean activityRequired);

    boolean isCarDockRequired();

    void setCarDockRequired(boolean carDockRequired);

    void setDisableBluetoothOnPower(boolean disableBluetoothOnPower);

    void setEnableBluetoothOnPower(boolean enableBluetoothOnPower);

    void setDisableScreenTimeout(boolean screenTimeout);

    void setAdjustVolumeLevel(boolean adjustVolumeLevel);

    void setMediaVolumeLevel(int mediaVolumeLevel);

    void setCallVolumeLevel(int level);

    void setEnableBluetooth(boolean bluetooth);

    void setBrightness(String brightness);

    void setAutoSpeaker(boolean autoSpeaker);

    String getDisableWifi();

    void setDisableWifi(String disableWifi);

    boolean isActivateCarMode();

    void setActivateCarMode(boolean activate);

    String getAutoAnswer();

    void setAutoAnswer(String autoAnswer);

    ComponentName getAutorunApp();

    void setAutorunApp(ComponentName autorunApp);

    boolean isSamsungDrivingMode();

    void setSamsungDrivingMode(boolean samsungDrivingMode);

    boolean isDisableScreenTimeoutCharging();

    void setDisableScreenTimeoutCharging(boolean disableScreenTimeoutCharging);

    int getScreenOrientation();

    void setScreenOrientation(int screenOrientation);

    boolean isHotspotOn();

    void setHotspotOn(boolean on);
}

File: app/src/main/java/com/anod/car/home/prefs/model/InCarInterface.kt
package com.anod.car.home.prefs.model

import android.content.ComponentName
import androidx.collection.ArrayMap
import com.anod.car.home.incar.ScreenOnAlert

/**
 * @author algavris
 * @date 08/04/2016.
 */
interface InCarInterface {
    var isInCarEnabled: Boolean
    var isAutoSpeaker: Boolean
    var btDevices: ArrayMap<String, String>
    var isPowerRequired: Boolean
    var isHeadsetRequired: Boolean
    val isBluetoothRequired: Boolean
    var isDisableBluetoothOnPower: Boolean
    var isEnableBluetoothOnPower: Boolean
    var isDisableScreenTimeout: Boolean
    var isAdjustVolumeLevel: Boolean
    var isActivityRequired: Boolean
    var mediaVolumeLevel: Int
    var callVolumeLevel: Int
    var isEnableBluetooth: Boolean
    var brightness: String
    var isCarDockRequired: Boolean
    var disableWifi: String
    var isActivateCarMode: Boolean
    var autoAnswer: String
    var autorunApp: ComponentName?
    var isSamsungDrivingMode: Boolean
    var isDisableScreenTimeoutCharging: Boolean
    var screenOrientation: Int
    var isHotspotOn: Boolean
    var screenOnAlert: ScreenOnAlert.Settings

    companion object {
        const val BRIGHTNESS_DISABLED = "disabled"
        const val BRIGHTNESS_AUTO = "auto"
        const val BRIGHTNESS_DAY = "day"
        const val BRIGHTNESS_NIGHT = "night"
        const val AUTOANSWER_DISABLED = "disabled"
        const val AUTOANSWER_IMMEDIATLY = "immediately"
        const val AUTOANSWER_DELAY_5 = "delay-5"
        const val WIFI_NOACTION = "no_action"
        const val WIFI_TURNOFF = "turn_off_wifi"
        const val WIFI_DISABLE = "disable_wifi"
        const val DEFAULT_VOLUME_LEVEL = 80
    }
}
